### PR TITLE
Homework Week3 Part 1. Mealsharing api

### PR DESCRIPTION
**Pull request by Lucha2040**

### CODE REVIEW

File: nodejs/week3/src/backend/api/meals.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
+const express = require("express");
+const app = express();
```
Comment (Hatko) [inline nodejs/week3/src/backend/api/meals.js]: Seems to be unused

File: nodejs/week3/src/backend/api/meals.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+// Returns all meals	
+router.get("/", (request, response) => {
+  pool.query("SELECT * FROM meal", function(error, results) {
+    if (error) {
+      console.log(error);
+      response.send("No meals were found, please check message error")
+    } else {
+    return response.json(results);
+    
+} next();
```
Comment (Hatko) [inline nodejs/week3/src/backend/api/meals.js]: `next` would crash if you ever reached this code. It's still after `if/else` blocks, so you will never have it executed. Could you fix it? Also, the alignment is broken in a lot of places. If you fix it, the code will become much more readable.

File: nodejs/week3/src/backend/api/meals.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+}
+) 
+});
+
+
+// Adds a new meal	
+router.post("/", (request, response) => {
+  const {title, description, location, when, max_reservations, price, created_date } = request.body;
+  const data = { 
+  title: title,
```
Comment (Hatko) [inline nodejs/week3/src/backend/api/meals.js]: You can use short syntax here

File: nodejs/week3/src/backend/api/meals.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+  when: when, 
+  max_reservations: max_reservations, 
+  price: price, 
+  created_date: created_date
+}   
+  pool.query("INSERT INTO meal SET ?", data, function(error, results, fields) {
+    if (error) {
+      response.send("No meals were found, please check message error")
+    } else {
+      response.json({...data, id:results.insertId}); //this line merges the data with the ID
```
Comment (Hatko) [inline nodejs/week3/src/backend/api/meals.js]: 👍

File: nodejs/week3/src/backend/api/meals.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+    console.log(results)
+    return response.json(results);
+  })
+});
+
+//Updates the meal by id	
+router.put("/", (request, response) => {
+  const {id, title, description, location, when, max_reservations, price, created_date } = request.body;
+  const data = { 
+  id: id,
```
Comment (Hatko) [inline nodejs/week3/src/backend/api/meals.js]: You can use short syntax here

File: nodejs/week3/src/backend/api/meals.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+});
+
+//Deletes the meal by id	
+router.delete("/", (request, response) => {
+  const id = request.body; 
+  pool.query("DELETE * FROM meal WHERE ?", id, function(error, results) {
+    if (error) {
+      response.send("Something went wrong, the meal was not delete")
+    }
+    response.json(request);
```
Comment (Hatko) [inline nodejs/week3/src/backend/api/meals.js]: You will call this even if you have an error 🙂

File: nodejs/week3/src/backend/api/reservations.js [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+  }
+  ) 
+  });
+  
+  
+  // Adds a new reservations	
+  router.post("/", (request, response) => {
+    const { Number_of_guests, Meal_id, Created_date } = request.body;
+    const data = { 
+    Number_of_guests: Number_of_guests,
```
Comment (Hatko) [inline nodejs/week3/src/backend/api/reservations.js]: Any reason for using this naming style?

File: nodejs/week3/src/backend/api/reservations.js [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+  // Returns reservations by id	
+  router.get("/", (request, response) => {
+    const id = request.body;  
+    pool.query("SELECT * FROM reservation WHERE ?", id, function(error, results) {
+      if (error) {
+        console.log(error)
+        response.send("No reservations were found, please check that the id is correct")
+      }
+      console.log(results)
+      return response.json(results);
```
Comment (Hatko) [inline nodejs/week3/src/backend/api/reservations.js]: There is no reason to return something from this function

Comment (Hatko) [reply nodejs/week3/src/backend/api/reservations.js]: You will call this code even if `pool.query` throws

File: nodejs/week3/src/backend/api/meals.js (hunk 1 above)
Comment (Lucha2040) [reply nodejs/week3/src/backend/api/meals.js]: Thanks!. Corrected in PR: https://github.com/Lucha2040/hyf-homework/pull/36/commits/586da540138991a0d004e8f01f601105b8d20d4d

File: nodejs/week3/src/backend/api/meals.js (hunk 2 above)
Comment (Lucha2040) [reply nodejs/week3/src/backend/api/meals.js]: Corrected in PR https://github.com/Lucha2040/hyf-homework/pull/36/commits/586da540138991a0d004e8f01f601105b8d20d4d
But some routing issues persist.

File: nodejs/week3/src/backend/api/meals.js (hunk 3 above)
Comment (Lucha2040) [reply nodejs/week3/src/backend/api/meals.js]: Corrected en next PR. 
Thanks!

File: nodejs/week3/src/backend/api/reservations.js (hunk 7 above)
Comment (Lucha2040) [reply nodejs/week3/src/backend/api/reservations.js]: Modify in last commit 0c7728991bd1f785fb8a2affa641961574b92c4e. 
Thanks for your time!

**Pull request merged**